File: 013-RomanToInteger/map.cpp
class Solution {
public:
    int romanToInt(std::string s)
    {
        int sum = 0;
        int start = 0;

        char char_arr[] = {'I', 'V', 'X', 'L', 'C', 'D', 'M'};
        int int_arr[] = {1, 5, 10, 50, 100, 500, 1000};

        std::map<char, int> roman_map;
        int map_len = sizeof(char_arr) / sizeof(char);

        for (int i = 0; i< map_len; ++i)
        {
            roman_map.insert(std::pair<char, int> (char_arr[i], int_arr[i]));
        }
        for (int i = 0; i < s.length() - 1; ++i)
        {
            if (roman_map[s[i]]>=roman_map[s[i + 1]])
            {
                sum += roman_map[s[i]];
            }
            else
            {

                sum -= roman_map[s[i]];
            }
        }
        sum += roman_map[s[s.length()-1]];
        return sum;
    }

File: 013-RomanToInteger/map.py
class Solution:
# @return an integer
def romanToInt(self, s):
    map={"I":1,"V":5,"X":10,"L":50,"C":100,"D":500,"M":1000}
    sum=0
    for i in range(0,len(s)):
        sum+=map[s[i]]     #s[i]

        if i>0 and map[s[i-1]] < map[s[i]]:
            sum=sum-2*map[s[i-1]]

    return sum 
